pkg/chartverifier/profiles/profile_test.go

```diff
@@ -4,80 +4,131 @@ import (
 	"fmt"
 	"github.com/google/go-cmp/cmp"
 	"github.com/redhat-certification/chart-verifier/pkg/chartverifier/checks"
+	"github.com/spf13/viper"
 	"github.com/stretchr/testify/assert"
 	"path/filepath"
 	"reflect"
 	"runtime"
 	"testing"
 )
 
+const (
+	NoVersion       string     = ""
+	configVersion10 string     = "1.0"
+	configVersion11 string     = "1.1"
+	configVersion12 string     = "1.2"
+	checkVersion10  string     = CheckVersion10
+	checkVersion11  string     = "v1.1"
+	NoVendorType    VendorType = ""
+)
+
 func TestProfile(t *testing.T) {
 
 	testProfile := getDefaultProfile("test")
-	testProfile.Name = "profile-1.0.0"
+	testProfile.Name = "profile-partner-1.1"
+	config := viper.New()
 
 	t.Run("Profile read from disk should match test profile", func(t *testing.T) {
-		diskProfile := Get()
+		diskProfile := New("1.1.0", config)
```

Comment:
This should be "1.1" right? I'm kind of confused by the naming of `configVersion`, `checkVersion`, `profileVersion` and `cmd.Version`..

Reply:
cmd.Version is the chart-verifier version - it is major.minor.patch, if we publish a new chart verifier with bug fixes we update the patch level, if we change behavior or add significant function we bump the minor version.

checkVersion is the version of a check, I didn't want to get into patch versions because we would get into check version hell - consider 1.0 as 1.0.any. if we update a check in a way which it can change an outcome for the same input we would bump it to 1.1 etc. Also conisder in terms of patch levels, we would always want the latest patch version so being able to specify older ones is redundant.

profileVersion - my original  intent was the profile version would mirror the chart-verifier version and we would issue a new one with each minor version update of the chart verifier. This breaks down and they need to be independent but the logic applied to checks, also applies here. 

configVersion is a profile version set through command line config when invoking the verifier. Maybe I could have picked a better name, but maybe not.

func New(version string, config *viper.Viper) *Profile - this function doesn't care whether it gets two digits or three, it just ignores the third digit. I did it this way because when the verify command calls New and it has a 3 digit version and it didn't want it to include logic to strip the 3rd digit. As a result the test sends 3 digits too. 

Hope this helps.